### PR TITLE
tests: add alsa interface spread test

### PR DESCRIPTION
**Pull request by fgimenez**

### CODE REVIEW

File: tests/main/interfaces-alsa/task.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+summary: Ensure that the alsa interface works.
+
+systems: [-debian-*]
```
Comment (morphis) [inline tests/main/interfaces-alsa/task.yaml]: What is preventing this test case from working on Debian?

File: tests/main/interfaces-alsa/task.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    A snap declaring a plug on this interface must be able to access to raw ALSA
+    devices, for this test we check the low level security rules by creating
+    specific devices under /dev/snd, a cgroup if needed and the ALSA state dir,
+    exercising in each caase the read or read-write permissions that must be in
+    place.
+
+prepare: |
+    echo "Given a snap declaring a plug on the alsa interface is installed"
+    . $TESTSLIB/snaps.sh
+    install_generic_consumer alsa
```
Comment (morphis) [inline tests/main/interfaces-alsa/task.yaml]: Btw. we have a alsa-utils snap in the which we could exercise here too.

File: tests/main/interfaces-alsa/task.yaml (hunk 1 above)
Comment (fgimenez) [inline tests/main/interfaces-alsa/task.yaml]: The test failed on the first run https://travis-ci.org/snapcore/snapd/builds/237202197#L1190 this check is with the plug disconnected https://github.com/snapcore/snapd/pull/3408/files#diff-38deae1fd3338b606e72783f027800a2R63, seems like the confinement is not working?

File: tests/main/interfaces-alsa/task.yaml (hunk 2 above)
Comment (fgimenez) [inline tests/main/interfaces-alsa/task.yaml]: Thanks! I'll take a look, we try to use as much as possible test snaps with code in snapd's tree, we can easily adapt them to the evolving conditions and upload to the store as needed. Also, it is good to use the smallest test snap possible to check the features exercised by the test, in this particular case we can use the tiniest consumer possible, generic-consumer https://github.com/snapcore/snapd/tree/master/tests/lib/snaps/generic-consumer

File: tests/main/interfaces-alsa/task.yaml (hunk 1 above)
Comment (morphis) [inline tests/main/interfaces-alsa/task.yaml]: Yes, there is no confinement on debian. That is what I introduced the `snap debug confinement = "strict"` check to all other interface tests to split them into two portions, one for the allowed part (as there is still seccomp in place and a few other things worth to check) and one for the disallowed path. The disallowed part is always skipped when `snap debug confinement` returns `none`. See https://github.com/snapcore/snapd/blob/master/tests/main/interfaces-firewall-control/task.yaml#L72 for an example.

Comment (fgimenez) [inline tests/main/interfaces-alsa/task.yaml]: Cool thanks a lot! Pushed changes for this, pls take a look

File: tests/main/interfaces-alsa/task.yaml (hunk 2 above)
Comment (morphis) [inline tests/main/interfaces-alsa/task.yaml]: Sounds good!

File: tests/main/interfaces-alsa/task.yaml (hunk 1 above)
Comment (morphis) [inline tests/main/interfaces-alsa/task.yaml]: LGTM

File: tests/main/interfaces-alsa/task.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+restore: |
+    rm -f *.error /dev/snd/mysnd-dev
```
Comment (sergiocazzolato) [inline tests/main/interfaces-alsa/task.yaml]: Should be removed the snaps installed in the prepare?

Comment (fgimenez) [inline tests/main/interfaces-alsa/task.yaml]: We don't need to remove the installed snaps, the whole initial state is restored before each task of the main suite https://github.com/snapcore/snapd/blob/master/spread.yaml#L300

Comment (sergiocazzolato) [inline tests/main/interfaces-alsa/task.yaml]: ah, good, thanks for the data

File: tests/main/interfaces-alsa/task.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    echo "When the plug is disconnected"
+    snap disconnect generic-consumer:alsa
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "The snap is not able to access snd devices"
+    if generic-consumer.cmd touch /dev/snd/mysnd-dev 2>snd-create.error; then
+        echo "Create snd device with disconnected plug should fail"
+        exit 1
+    fi
+    MATCH "Permission denied" ./snd-create.error
```
Comment (niemeyer) [inline tests/main/interfaces-alsa/task.yaml]: That won't work at the moment. If the matching fails, it won't print the file content.

That said, let's leave it as is and fix MATCH in spread instead.

**Pull request merged**